pkg/skaffold/build/jib/sync.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jib
+
+import (
+	"bytes"
+	"encoding/json"
+	"os"
+	"os/exec"
+	"regexp"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+type SyncMap map[string]SyncEntry
+
+type SyncEntry struct {
+	Dest     []string
+	FileTime time.Time
+	IsDirect bool
+}
+
+type JSONSyncMap struct {
+	Direct    []JSONSyncEntry `json:"direct"`
+	Generated []JSONSyncEntry `json:"generated"`
+}
+
+type JSONSyncEntry struct {
+	Src  string `json:"src"`
+	Dest string `json:"dest"`
+}
+
+func getSyncMapFromSystem(cmd *exec.Cmd) (*SyncMap, error) {
+	jsm := JSONSyncMap{}
+	stdout, err := util.RunCmdOut(cmd)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get Jib sync map")
+	}
+
+	// To parse the output, search for "BEGIN JIB JSON", then unmarshal the next line into the pathMap struct.
+	// Syncmap is transitioning to "BEGIN JIB JSON: SYNCMAP/1" starting in jib 2.0.0
+	// perhaps this feature should only be included from 2.0.0 onwards? And we generally avoid this?
```

Comment:
are there other differences in the sync output or it's the same?

Reply:
The output is the same, we decided we wanted some more matching data. All our json is prefixed with "BEGIN JIB JSON" so if we were run two of these in one execution (for example: getDependencies and getSyncMap) we wouldn't be able to know which json is for which easily.

Reply:
The difference is in that 1.8.0 we already release syncmap generation with the prefix "BEGIN JIB JSON", and we're correcting that with 2.0.0 -> "BEGIN JIB JSON: SYNCMAP/1"

Reply:
Sounds good. This should be fine then to include from 1.8.0, I see no issue there. Do you see any risk there? I guess it would make this expression cleaner, but other than that there is no advantage to limit to 2.0.0+, right?

Reply:
I think we can limit sync to 2.0.0+ and just remove this code. 2.0.0 should go out next week

Reply:
It does kind of need to be here for now for this to "work" while in development.

Reply:
we can just remove it once 2.0.0 is out, let's merge it.